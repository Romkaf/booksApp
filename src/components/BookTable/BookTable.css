.root {
	width: 100%;
	margin-bottom: 1rem;
	border-collapse: collapse;
}

.wrapper {
	overflow: hidden;
}

.row {
	display: table-header-group;
	vertical-align: middle;
	border-color: inherit;
}

.ceil {
	width: 25%;
	padding: 0.75rem;
	border-bottom: 2px solid var(--grey);
	border-top: 1px solid var(--grey);
	vertical-align: bottom;
	text-align: inherit;
}

@media (max-width: 500px) {
	.ceil {
		padding: 0.15rem;
		font-size: 16px;
	}
}
