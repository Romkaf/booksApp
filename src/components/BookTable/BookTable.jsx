import React, { useEffect } from 'react';
import Row from '@components/BookRow';
import { useSelector, useDispatch } from 'react-redux';
import { fetchBooks } from '@models/actions';
import styles from './BookTable.css';

const BookTable = () => {
	const { root, row, ceil } = styles;
	const { books, loading, error } = useSelector((state) => state.books);
	const dispatch = useDispatch();

	useEffect(() => dispatch(fetchBooks()), []);

	if (loading) {
		return <h1>Loading...</h1>;
	}

	if (error) {
		return <h2>{error}</h2>;
	}

	return (
		<div>
			<table className={root}>
				<thead className={row}>
					<tr>
						<th className={ceil}>Title</th>
						<th className={ceil}>Authors</th>
						<th className={ceil}>Publisher</th>
						<th className={ceil}>Date</th>
					</tr>
				</thead>

				<tbody>
					{books.map((it) => (
						<Row item={it} key={it.id} />
					))}
				</tbody>
			</table>
		</div>
	);
};

export default BookTable;
