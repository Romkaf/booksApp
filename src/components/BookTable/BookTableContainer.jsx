import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchBooks } from '@models/actions';
import BookTable from './BookTable';

const BookTableContainer = () => {
	const {
		books: { books, loading },
		filters,
		pagination,
	} = useSelector((state) => state);

	const dispatch = useDispatch();

	useEffect(() => dispatch(fetchBooks({ ...filters, ...pagination })), [
		filters,
		pagination,
	]);

	return <BookTable books={books} loading={loading} />;
};

export default BookTableContainer;
