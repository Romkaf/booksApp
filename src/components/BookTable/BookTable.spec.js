import React from 'react';
import BookTable from './BookTable';
import { shallow } from 'enzyme';

describe('BookTable', () => {
	it('should render component', () => {
		const component = shallow(<BookTable />);
		expect(component).toMatchSnapshot();
	});
});
