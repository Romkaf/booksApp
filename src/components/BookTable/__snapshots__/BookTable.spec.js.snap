// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookTable should render component 1`] = `
<div
  className="wrapper"
>
  <table
    className="root"
  >
    <Memo(TableHead) />
    <TableBodyContainer />
  </table>
  <PaginationContainer />
</div>
`;
